src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -144,16 +155,42 @@ public Collector newCollector() {
     }
 
     private Collector getHybridQueryCollector() {
-        if (sortAndFormats == null) {
-            return new HybridTopScoreDocCollector(numHits, hitsThresholdChecker);
+        if (collapseContext != null) {
```

Comment:
we need a factory for hybrid query collectors. Can you draft something simple - pack all needed class fields into DTO class and copy logic from this `getCollector` method to `createCollector` (or pick other name) method of that factory.